components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultDomainObjectFactoryTest.java

```diff
@@ -0,0 +1,255 @@
+        DocumentReference patientDocument = mock(DocumentReference.class);
+        DocumentReference creatorDocument = mock(DocumentReference.class);
+        UriBuilder uriBuilder = mock(UriBuilder.class);
+        URI uri = new URI("uri");
+        Date creationDate = new Date(1);
+
+        when(this.patient.getDocument()).thenReturn(patientDocument);
+        when(this.access.hasAccess(Right.VIEW, null, patientDocument)).thenReturn(true);
+        when(this.documentAccessBridge.getDocument(patientDocument)).thenReturn(documentReference);
+        when(this.patient.getId()).thenReturn("id");
+        when(this.patient.getExternalId()).thenReturn("externalid");
+        when(this.patient.getReporter()).thenReturn(creatorDocument);
+
+        when(creatorDocument.toString()).thenReturn("creator");
+        when(documentReference.getAuthorReference()).thenReturn(creatorDocument);
+        when(documentReference.getVersion()).thenReturn("version");
+        when(documentReference.getCreationDate()).thenReturn(creationDate);
+        when(documentReference.getDate()).thenReturn(creationDate);
+        when(this.uriInfo.getBaseUriBuilder()).thenReturn(uriBuilder);
+        when(uriBuilder.path(PatientResource.class)).thenReturn(uriBuilder);
+        when(uriBuilder.build("id")).thenReturn(uri);
+
+        PatientSummary patientSummary = mocker.getComponentUnderTest().createPatientSummary(this.patient, this.uriInfo);
+
+        assertEquals("id", patientSummary.getId());
+        assertEquals("externalid", patientSummary.getEid());
+        assertEquals("creator", patientSummary.getCreatedBy());
+        assertEquals("creator", patientSummary.getLastModifiedBy());
+        assertEquals("version", patientSummary.getVersion());
+        assertTrue(patientSummary.getCreatedOn() instanceof DateTime);
+        assertTrue(patientSummary.getLastModifiedOn() instanceof DateTime);
+        assertEquals(1, patientSummary.getLinks().size());
+        assertEquals("uri", patientSummary.getLinks().get(0).getHref());
+    }
+
+    @Test
+    public void createPatientFromSummaryWrongLength() throws Exception
+    {
+        Object[] summary = {new Object()};
+        assertNull(this.mocker.getComponentUnderTest().createPatientSummary(summary, this.uriInfo));
+    }
+
+    @Test
+    public void createPatientFromSummaryWrongObjectTypes() throws Exception
+    {
+        Object[] summary = {new Object(), new Object(), new Object(), new Object(), new Object(), new Object(), new Object()};
+        assertNull(this.mocker.getComponentUnderTest().createPatientSummary(summary, this.uriInfo));
+    }
+
+    @Test
+    public void createPatientFromSummaryNoAccess() throws Exception
+    {
+        when(this.patient.getDocument()).thenReturn(null);
+        when(this.access.hasAccess(Right.VIEW, null, null)).thenReturn(false);
+        assertNull(this.mocker.getComponentUnderTest().createPatientSummary(this.patient, this.uriInfo));
```

Comment:
This test is currently redundant with `cannotCreatePatientWithNoAccess()`
The first line of this method should be changed to `when(this.stringResolver.resolve(Matchers.anyString())).thenReturn(null);`
and the last line should be changed to `assertNull(this.mocker.getComponentUnderTest().createPatientSummary({some array of objects}, this.uriInfo));`